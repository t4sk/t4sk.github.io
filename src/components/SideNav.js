import React from "react"
import PropTypes from "prop-types"
import { List } from "semantic-ui-react"
import styles from "./SideNav.module.css"

export function SideNav(props) {
  const { baseUrl, chapters } = props

  // TODO active
  // NOTE: using Link does not load highlight.js so he use anchor to navigate
  // TODO FIX - use react router, load syntax highlight, reset scroll by document.getElementById("root").scrollTo(0, 0)
  return (
    <div className={styles.component}>
      <div className={styles.sideNav}>
        {chapters.map(({ header, lessons }) => (
          <div key={header}>
            <h2>{header}</h2>
            <List divided relaxed selection>
              {lessons.map((lesson, i) => (
                <List.Item key={i}>
                  <List.Icon
                    name={lesson.icon}
                    {...(lesson.color ? { color: lesson.color } : {})}
                    size="large"
                    verticalAlign="middle"
                  />
                  <List.Content>
                    <List.Header as="a" href={`${baseUrl}/${lesson.path}`}>
                      {lesson.header}
                    </List.Header>
                    {lesson.description && (
                      <List.Description>{lesson.description}</List.Description>
                    )}
                  </List.Content>
                </List.Item>
              ))}
            </List>
          </div>
        ))}
      </div>
    </div>
  )
}

SideNav.propTypes = {
  baseUrl: PropTypes.string.isRequired,
  chapters: PropTypes.arrayOf(
    PropTypes.shape({
      header: PropTypes.string.isRequired,
      lessons: PropTypes.arrayOf(
        PropTypes.shape({
          icon: PropTypes.string.isRequired,
          color: PropTypes.string,
          path: PropTypes.string,
          header: PropTypes.string.isRequired,
          description: PropTypes.string,
        })
      ).isRequired,
    })
  ).isRequired,
}

export default SideNav
