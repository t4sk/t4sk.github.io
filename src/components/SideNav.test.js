import React from "react"
import { shallow } from "enzyme"
import { SideNav } from "./SideNav"

const props = {
  baseUrl: "http://localhost:3000",
  chapters: [
    {
      header: "Solidity Contract",
      lessons: [
        {
          icon: "youtube",
          color: "red",
          path: "00-overview",
          header: "Intro",
          description: "",
        },
        {
          icon: "code",
          color: "",
          path: "01-constructor",
          header: "Constructor",
          description: "Exercise",
        },
        {
          icon: "github",
          color: "",
          path: "",
          header: "Code",
        },
      ],
    },
    {
      header: "Truffle",
      lessons: [
        {
          icon: "youtube",
          color: "red",
          path: "00-overview",
          header: "Intro",
          description: "",
        },
        {
          icon: "code",
          color: "",
          path: "01-constructor",
          header: "Constructor",
          description: "Exercise",
        },
      ],
    },
  ],
}

beforeEach(() => {})

test("it renders", () => {
  const component = shallow(<SideNav {...props} />)

  expect(component).toMatchSnapshot()
})
