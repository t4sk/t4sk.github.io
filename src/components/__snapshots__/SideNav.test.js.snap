// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="component"
>
  <div
    className="sideNav"
  >
    <div
      key="Solidity Contract"
    >
      <h2>
        Solidity Contract
      </h2>
      <List
        divided={true}
        relaxed={true}
        selection={true}
      >
        <ListItem
          key="0"
        >
          <ListIcon
            color="red"
            name="youtube"
            size="large"
            verticalAlign="middle"
          />
          <ListContent>
            <ListHeader
              as="a"
              href="http://localhost:3000/00-overview"
            >
              Intro
            </ListHeader>
          </ListContent>
        </ListItem>
        <ListItem
          key="1"
        >
          <ListIcon
            name="code"
            size="large"
            verticalAlign="middle"
          />
          <ListContent>
            <ListHeader
              as="a"
              href="http://localhost:3000/01-constructor"
            >
              Constructor
            </ListHeader>
            <ListDescription>
              Exercise
            </ListDescription>
          </ListContent>
        </ListItem>
        <ListItem
          key="2"
        >
          <ListIcon
            name="github"
            size="large"
            verticalAlign="middle"
          />
          <ListContent>
            <ListHeader
              as="a"
              href="http://localhost:3000/"
            >
              Code
            </ListHeader>
          </ListContent>
        </ListItem>
      </List>
    </div>
    <div
      key="Truffle"
    >
      <h2>
        Truffle
      </h2>
      <List
        divided={true}
        relaxed={true}
        selection={true}
      >
        <ListItem
          key="0"
        >
          <ListIcon
            color="red"
            name="youtube"
            size="large"
            verticalAlign="middle"
          />
          <ListContent>
            <ListHeader
              as="a"
              href="http://localhost:3000/00-overview"
            >
              Intro
            </ListHeader>
          </ListContent>
        </ListItem>
        <ListItem
          key="1"
        >
          <ListIcon
            name="code"
            size="large"
            verticalAlign="middle"
          />
          <ListContent>
            <ListHeader
              as="a"
              href="http://localhost:3000/01-constructor"
            >
              Constructor
            </ListHeader>
            <ListDescription>
              Exercise
            </ListDescription>
          </ListContent>
        </ListItem>
      </List>
    </div>
  </div>
</div>
`;
